### PR TITLE
Improving radioButtons accessibility: Closing #3105

### PR DESCRIPTION
**Pull request by jooyoungseo**
This PR addresses #3105.

After this patch, ...:

1. Choices generated from `radioButtons()` are semantically grouped together for assistive technologies (e.g., screen readers).
1. Group-level label passed to `label` argument in `radioButtons()` is now tied to each choice, meaning that screen readers announce the group label along with each radio option name when tabbing around.

Test has been carried out with various screen readers, giving satisfactory results.

Note that there is no break for other Shiny functionalities, nor does it slowdown Shiny performance. (it simply improves the accessibility).

## Reprex

``` r
library(shiny)
ui <- fluidPage(
  radioButtons("dist", "Distribution type:",
               c("Normal" = "norm",
                 "Uniform" = "unif",
                 "Log-normal" = "lnorm",
                 "Exponential" = "exp")),
  plotOutput("distPlot")
)

server <- function(input, output) {
  output$distPlot <- renderPlot({
    dist <- switch(input$dist,
                   norm = rnorm,
                   unif = runif,
                   lnorm = rlnorm,
                   exp = rexp,
                   rnorm)

    hist(dist(500))
  })
}

shinyApp(ui, server)
```

### CODE REVIEW

**Comment by cpsievert** [comment]
Thanks @jooyoungseo! I imagine we should also be making similar changes to `checkboxGroupInput()`?

**Comment by jooyoungseo** [comment]
@cpsievert, no problem!

Yes, I think we may want to apply the same patch to the checkbox group, which I need to verify later on.

What about we open that PR separately so that I will be able to pull and test?

**Comment by cpsievert** [comment]
@jooyoungseo sounds good, thanks

**Review by cpsievert — APPROVED** [review]

File: R/input-radiobuttons.R [hunk 1]
```diff
@@ -108,6 +108,11 @@ radioButtons <- function(inputId, label, choices = NULL, selected = NULL,
     style = if (!is.null(width)) paste0("width: ", validateCssUnit(width), ";"),
     class = divClass,
     shinyInputLabel(inputId, label),
+    # Improve accessibility (#3105)
+    # https://www.w3.org/TR/2017/WD-wai-aria-practices-1.1-20170628/examples/radio/radio-1/radio-1.html
+    role = "radiogroup",
+    # The `aria-labelledby` is tied to id value in `shinyInputLabel()` to be announced to screen readers and in Braille:
+    `aria-labelledby` = paste0(inputId, "-label"),
```
Comment (wch) [inline R/input-radiobuttons.R:115]: Probably safer to extract this value from the object returned by `shinyInputLabel(inputId, label)`

**Comment by cpsievert** [comment]
Thanks @jooyoungseo! We'll take this over the finish line in https://github.com/rstudio/shiny/pull/3186

**Pull request closed without merging**